activerecord/lib/active_record/attributes.rb

```diff
@@ -102,6 +102,12 @@ def persistable_attribute_names # :nodoc:
         @persistable_attribute_names ||= connection.schema_cache.columns_hash(table_name).keys
       end
 
+      # Returns an array of arel attributes to be used in select statements
+      # when the field list is not specified.
+      def default_select_columns
```

Comment:
You could define this as:

```ruby
class_attribute default_select_columns
self.default_select_columns = [arel_table[Arel.star]]
```

User API would then be:

```ruby
class User
  self.default_select_columns = [...]
```

And if this is user API, should document both getter and setter use cases, as well as explain in docs what you explained in this PR.